Source/Hat.cs

```diff
@@ -132,35 +144,45 @@ private void LogModLoadingState(Mod mod, bool loadState)
             }
         }
 
-        private void RemoveDuplicates()
+        private void DisableMods()
+        {
+            foreach (Mod mod in Mods)
+            {
+                ModConfig config = ConfigHelper.GetModConfig(mod.Info.Name, mod.Info.Version);
+                if (config.Disabled.HasValue && config.Disabled.Value == true)
+                    mod.IsEnabled = false;
+            }
+        }
+
+        private void DisableDuplicates()
```

Comment:
We have to consider how having multiple versions of the same mod in the game's directory will affect the mod loader, and how it should be communicated properly to the end user. Something like a slide field for version for mods with multiple versions might be appreciated. It could still use the "enabled/disabled" system but it's best not to clutter the mod menu.

Reply:
Sure thing. Also, I revert my previous statement and I think that actually having something like `SelectedVersion` as an optional parameter would be a nice alternative to storing a separate record for each installed version of the mod - this way it can default to the newest one when the option is not present or it points to non-existing version of the mod.